eng/intellisense.targets

```diff
@@ -36,15 +35,13 @@
     </ItemGroup>
   </Target>
 
-  <!-- Copy the intellisense file to the folder we use to produce the targeting pack.
-       The condition is for the RID agnostic build of source assemblies that are part of the .NET Core shared framework. -->
-  <Target Name="CopyDocumentationFileToXmlDocDir"
-          AfterTargets="CopyFilesToOutputDirectory"
-          Condition="'$(IsNetCoreAppSrc)' == 'true' and '$(TargetFramework)' == '$(NetCoreAppCurrent)'">
```

Comment:
Which runtime-specific target-framework do we end up shipping the ref-pack build from -- would that be Windows?  I wonder if there is any case when the windows thing is less good than the non-windows thing.

To double check your change - it might be interesting to look at a diff of whats in the ref-pack before and after this change to confirm that it's a net-positive fix.  I think it should be, but it'd be good to check to see if there's anything we are missing.

Reply:
Yes, the targeting pack that ships to customers is produced in the `Windows_AllConfigurations` leg. 

When comparing main vs with this change, the number of intellisense files slightly change because we don't copy files for private assemblies anymore (as those aren't used by the ref pack): https://www.diffchecker.com/6Y0xmGmm/

When comparing file contents, they are completely identical: https://gist.github.com/ViktorHofer/0a1d0d8d2357238e3aaf163a4aa56f79

FWIW these are the source-of-truth files in runtime:
- System.Runtime.Extensions.xml
- System.Threading.Tasks.xml
- System.Text.Encoding.xml
- System.Globalization.xml
- System.Resources.ResourceManager.xml
- System.Security.Principal.xml
- System.Diagnostics.Tools.xml
- System.AppContext.xml
- System.Globalization.Calendars.xml
- System.Security.Cryptography.Csp.xml
- System.IO.UnmanagedMemoryStream.xml
- System.Runtime.CompilerServices.Unsafe.xml
- System.Reflection.xml
- System.Diagnostics.Debug.xml
- System.Threading.Tasks.Extensions.xml
- System.Buffers.xml
- System.Threading.Timer.xml
- System.Xml.XmlDocument.xml
- System.Security.SecureString.xml
- System.Runtime.Handles.xml
- System.IO.FileSystem.Primitives.xml
- System.Security.Cryptography.Primitives.xml
- System.Reflection.Extensions.xml
- System.Security.Cryptography.Cng.xml
- System.Resources.Reader.xml
- System.IO.xml
- System.Security.Cryptography.OpenSsl.xml
- System.Globalization.Extensions.xml
- System.Security.Cryptography.X509Certificates.xml
- System.IO.FileSystem.xml
- System.ValueTuple.xml
- System.Dynamic.Runtime.xml
- System.Security.Cryptography.Algorithms.xml
- System.Runtime.InteropServices.RuntimeInformation.xml
- System.Data.DataSetExtensions.xml
- System.Security.Cryptography.Encoding.xml

Reply:
Thank you for doing this comparison.